src/main/java/world/bentobox/challenges/tasks/TryToComplete.java

```diff
@@ -423,239 +422,202 @@ private void fullFillRequirements(ChallengeResult result)
+        case INVENTORY_TYPE -> {
+            // If remove items, then remove them
+            if (this.getInventoryRequirements().isTakeItems()) {
+                int sumEverything = result.requiredItems.stream().
                         mapToInt(itemStack -> itemStack.getAmount() * result.getFactor()).
                         sum();
 
-                    Map<ItemStack, Integer> removedItems =
+                Map<ItemStack, Integer> removedItems =
                         this.removeItems(result.requiredItems, result.getFactor());
 
-                    int removedAmount = removedItems.values().stream().mapToInt(num -> num).sum();
+                int removedAmount = removedItems.values().stream().mapToInt(num -> num).sum();
 
-                    // Something is not removed.
-                    if (sumEverything != removedAmount)
-                    {
-                        Utils.sendMessage(this.user,
+                // Something is not removed.
+                if (sumEverything != removedAmount) {
+                    Utils.sendMessage(this.user,
                             this.world,
                             Constants.ERRORS + "cannot-remove-items");
 
-                        result.removedItems = removedItems;
-                        result.meetsRequirements = false;
-                    }
+                    result.removedItems = removedItems;
+                    result.meetsRequirements = false;
                 }
             }
-            case OTHER_TYPE -> {
-                OtherRequirements requirements = this.challenge.getRequirements();
+        }
+        case OTHER_TYPE -> {
+            OtherRequirements requirements = this.challenge.getRequirements();
 
-                if (this.addon.isEconomyProvided() && requirements.isTakeMoney())
-                {
-                    this.addon.getEconomyProvider().withdraw(this.user, requirements.getRequiredMoney());
-                }
+            if (this.addon.isEconomyProvided() && requirements.isTakeMoney()) {
+                this.addon.getEconomyProvider().withdraw(this.user, requirements.getRequiredMoney());
+            }
 
-                if (requirements.isTakeExperience() &&
-                    this.user.getPlayer().getGameMode() != GameMode.CREATIVE)
-                {
-                    // Cannot take anything from creative game mode.
-                    this.user.getPlayer().setTotalExperience(
+            if (requirements.isTakeExperience() && this.user.getPlayer().getGameMode() != GameMode.CREATIVE) {
+                // Cannot take anything from creative game mode.
+                this.user.getPlayer().setTotalExperience(
                         this.user.getPlayer().getTotalExperience() - requirements.getRequiredExperience());
-                }
             }
-            case STATISTIC_TYPE -> {
-                StatisticRequirements requirements = this.challenge.getRequirements();
+        }
+        case STATISTIC_TYPE -> {
```

Comment:
The main change in this class is here where multiple stats are checked instead of just the one.